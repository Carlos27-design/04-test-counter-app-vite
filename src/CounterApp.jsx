import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleSum = () => {
    // console.log(event);

    setCounter(counter + 1);
    // setCounter((c) => c + 1);
  };

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleSum}>Incrementar +</button>
      <button onClick={handleSubtract}>Disminuir -</button>
      <button aria-label='btn-reset' onClick={handleReset}>
        Resetear
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
