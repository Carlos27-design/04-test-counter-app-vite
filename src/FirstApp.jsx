import PropTypes from 'prop-types';

// const newMessage = {
//   message: 'Hola Mundo',
//   name: 'Carlos',
// };

//Se puede pero si la funcion no es asincrona
// const saludar = () => {
//   return 'Hola Mundo';
// };

// const mensaje = () => {
//   return 'Hola manito, Como estas Espero que estes bien no mÃ¡s ';
// };

// const numero = (a, b) => {
//   return a + b;
// };
export const FirstApp = ({
  title,
  subtitle = 'Todo el mundo explota po loco',
  name,
}) => {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      {/*<code>{JSON.stringify(newMessage)}</code>*/}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

// Esto ayuda para ver como funciona nuestro component por sus Propertys
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

// Estos son Props por defecto
FirstApp.defaultProps = {
  //title: 'No hay titulos',
  subtitle: 'No hay subtitulo',
  name: 'Carlos Coronado',
};
