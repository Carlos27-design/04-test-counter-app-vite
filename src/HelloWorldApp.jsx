export const HelloWorldApp = () => {
  return <h1>HelloWorldApp</h1>;
};

//Formas con menos codigo
//export const HelloWorldApp = () => <h1>Arrow Function</h1>;
