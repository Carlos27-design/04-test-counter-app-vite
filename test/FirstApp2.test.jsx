import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
  const title = 'Hola, soy Carlos Coronado';
  const subtitle = 'Soy un subtitulo';

  test('La primera pruena tiene que hacer match con el snapshop', () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hola, soy Carlos Coronado"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe de mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
