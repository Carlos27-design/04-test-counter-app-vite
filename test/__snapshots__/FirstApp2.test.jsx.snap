// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> La primera pruena tiene que hacer match con el snapshop 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Carlos Coronado
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Carlos Coronado
  </p>
</div>
`;
